samples/csharp/getting-started/Ranking_AutoML/Ranking/Ranking/Program.cs

```diff
@@ -0,0 +1,232 @@
+
+        static void Main(string[] args)
+        {
+            var mlContext = new MLContext(seed: 0);
+
+            // Create, train, evaluate and save a model
+            BuildTrainEvaluateAndSaveModel(mlContext);
+
+            // Make a single test prediction loading the model from .ZIP file
+            TestSinglePrediction(mlContext);
+
+            Console.WriteLine("=============== End of process, hit any key to finish ===============");
+            Console.ReadKey();
+        }
+
+        private static ITransformer BuildTrainEvaluateAndSaveModel(MLContext mlContext)
+        {
+            // STEP 1: Download and load the data
+            GetData(InputPath, OutputPath, TrainDatasetPath, TrainDatasetUrl, TestDatasetUrl, TestDatasetPath,
+                ValidationDatasetUrl, ValidationDatasetPath);
+
+            //ColumnInferenceResults columnInference = mlContext.Auto().InferColumns(TrainDatasetPath, labelColumnIndex: 0, 
+            //    separatorChar: '\t', hasHeader: true, groupColumns: false, allowSparse: true);
+
+            var textLoaderOptions = new TextLoader.Options
+            {
+                Separators = new[] { '\t' },
+                HasHeader = true,
+                Columns = new[]
+                {
+                    new TextLoader.Column("Label", DataKind.Single, 0),
+                    new TextLoader.Column("GroupId", DataKind.Int32, 1),
+                    new TextLoader.Column("Features", DataKind.Single, 2, 133),
+                }
+            };
+
+            TextLoader textLoader = mlContext.Data.CreateTextLoader(textLoaderOptions);
+            IDataView trainDataView = textLoader.Load(TrainDatasetPath);
+            IDataView validationDataView = textLoader.Load(ValidationDatasetPath);
+            IDataView testDataView = textLoader.Load(TestDatasetPath);
+
+            // STEP 2: Display first few rows of training data
+            ConsoleHelper.ShowDataViewInConsole(mlContext, trainDataView);
+
+            // STEP 3: Initialize our user-defined progress handler that AutoML will 
+            // invoke after each model it produces and evaluates.
+            var progressHandler = new RankingExperimentProgressHandler();
+
+            // STEP 4: Run AutoML ranking experiment
+            ConsoleHelper.ConsoleWriteHeader("=============== Running AutoML experiment ===============");
+            Console.WriteLine($"Running AutoML ranking experiment for {ExperimentTime} seconds...");
+
+            var experimentSettings = new RankingExperimentSettings
+            {
+                MaxExperimentTimeInSeconds = ExperimentTime,
+                OptimizingMetric = RankingMetric.Ndcg
+            };
+
+            ExperimentResult<RankingMetrics> experimentResult = mlContext.Auto()
+                .CreateRankingExperiment(experimentSettings)
+                .Execute(
+                    trainData: trainDataView,
+                    validationData: testDataView,
```

Comment:
Should be `validationDataView`:
```suggestion
                    validationData: validationDataView,
```